packages/base-driver/lib/basedriver/commands/find.js

```diff
@@ -1,51 +1,108 @@
-const commands = {}, helpers = {}, extensions = {};
+/* eslint-disable no-unused-vars */
+/* eslint-disable require-await */
+// @ts-check
+import {errors} from '../../protocol';
 
-
-// Override the following function for your own driver, and the rest is taken
-// care of!
-
-// helpers.findElOrEls = async function (strategy, selector, mult, context) {}
-//   strategy: locator strategy
-//   selector: the actual selector for finding an element
-//   mult: multiple elements or just one?
-//   context: finding an element from the root context? or starting from another element
-//
-// Returns an object which adheres to the way the JSON Wire Protocol represents elements:
-// { ELEMENT: # }    eg: { ELEMENT: 3 }  or { ELEMENT: 1.023 }
-
-helpers.findElOrElsWithProcessing = async function findElOrElsWithProcessing (strategy, selector, mult, context) {
-  this.validateLocatorStrategy(strategy);
-  try {
-    return await this.findElOrEls(strategy, selector, mult, context);
-  } catch (err) {
-    if (this.opts.printPageSourceOnFindFailure) {
-      const src = await this.getPageSource();
-      this.log.debug(`Error finding element${mult ? 's' : ''}: ${err.message}`);
-      this.log.debug(`Page source requested through 'printPageSourceOnFindFailure':`);
-      this.log.debug(src);
+/**
+ *
+ * @param {ReturnType<import('./event').EventMixin>} Base
+ * @returns {import('../driver').BaseDriverBase<import('@appium/types').TimeoutCommands & import('@appium/types').EventCommands & import('@appium/types').FindCommands>}
```

Comment:
this is awful

Reply:
can't we alias this type variable somewhere so it's not so ugly?

Reply:
We can create a bunch of typedefs.  In TS this would be a bunch of `import` statements.

Reply:
yeah, but imports are on their own up at the top, this is "in the middle" of code. it's ugly. typedefs are probably prettier.